akka-serialization-jackson/src/main/scala/akka/serialization/jackson/JacksonSerializer.scala

```diff
@@ -357,29 +360,48 @@ import com.fasterxml.jackson.dataformat.cbor.CBORFactory
   }
 
   def compress(bytes: Array[Byte]): Array[Byte] = {
-    val bos = new ByteArrayOutputStream(BufferSize)
-    val zip = new GZIPOutputStream(bos)
-    try zip.write(bytes)
-    finally zip.close()
-    bos.toByteArray
+
+    def compressWith(compress: OutputStream => OutputStream): Array[Byte] = {
+      val compressedOut = compress(new ByteArrayOutputStream(BufferSize))
+      try compressedOut.write(bytes)
+      finally compressedOut.close()
+      new ByteArrayOutputStream(BufferSize).toByteArray
+    }
+
+    algorithm match {
+      case Compression.OFF => bytes
+      case Compression.GZip if bytes.length <= compressLargerThan => bytes
+      case Compression.GZip => compressWith(new GZIPOutputStream(_))
+      case Compression.LZ4 => compressWith(new LZ4FrameOutputStream(_))
```

Comment:
Use `LZ4FrameOutputStream.BLOCKSIZE.SIZE_64KB` parameter.

JMH results are horrible with the default 4MB. I tried with same test data as in https://github.com/akka/akka/pull/27067

Size in bytes:
```
off:  5360
gzip: 3469
lz4:  4215
```

JMH results for `JacksonSerializationBench` with that data
```
[info] Benchmark                                                         (algo)  (serializerName)   Mode  Cnt         Score        Error   Units
[info] JacksonSerializationBench.large                                      off      jackson-json  thrpt    5     13480.766 ±   1889.994   ops/s
[info] JacksonSerializationBench.large:·gc.alloc.rate.norm                  off      jackson-json  thrpt    5     77881.103 ±     13.115    B/op

[info] JacksonSerializationBench.large                                     gzip      jackson-json  thrpt    5      4552.216 ±    310.308   ops/s
[info] JacksonSerializationBench.large:·gc.alloc.rate.norm                 gzip      jackson-json  thrpt    5    120508.622 ±     37.079    B/op

[info] JacksonSerializationBench.large                                      lz4      jackson-json  thrpt    5       454.367 ±     33.257   ops/s
[info] JacksonSerializationBench.large:·gc.alloc.rate.norm                  lz4      jackson-json  thrpt    5  16911932.851 ±    364.928    B/op
```

Switching to 64KB it's looking better

```
[info] Benchmark                                                     (algo)  (serializerName)   Mode  Cnt       Score       Error   Units
[info] JacksonSerializationBench.large                                  lz4      jackson-json  thrpt    5    9322.556 ±   992.144   ops/s
[info] JacksonSerializationBench.large:·gc.alloc.rate.norm              lz4      jackson-json  thrpt    5  380593.746 ±    17.155    B/op
```

Still generating pretty much garbage. Might be something more that can be tuned?